homeassistant/components/smart_meter_texas/sensor.py

```diff
@@ -0,0 +1,89 @@
+"""Support for Smart Meter Texas sensors."""
+import logging
+
+from smart_meter_texas import Meter
+
+from homeassistant.const import CONF_ADDRESS, ENERGY_KILO_WATT_HOUR
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, ELECTRIC_METER, ESIID, ICON, METER_NUMBER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Smart Meter Texas sensors."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    meters = coordinator.data.meters
+
+    async_add_entities(
+        [SmartMeterTexasSensor(meter, coordinator) for meter in meters], True
+    )
+
+
+class SmartMeterTexasSensor(Entity):
+    """Representation of an Smart Meter Texas sensor."""
+
+    def __init__(self, meter: Meter, coordinator: DataUpdateCoordinator):
+        """Initialize the sensor."""
+        self.meter = meter
+        self.coordinator = coordinator
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return ENERGY_KILO_WATT_HOUR
+
+    @property
+    def name(self):
+        """Device Name."""
+        return f"{ELECTRIC_METER} {self.meter.meter}"
+
+    @property
+    def icon(self):
+        """Device Ice."""
+        return ICON
+
+    @property
+    def unique_id(self):
+        """Device Uniqueid."""
+        return f"{METER_NUMBER}_{self.meter.meter}"
```

Comment:
```suggestion
        return self.meter.esiid
```

Should we use the ESI ID since its intended to be a unique identifier ?

Reply:
The ESIID seems to be a unique identifier for an address and not for a meter. Since an address could have multiple meters I believe the meter makes the most sense here.

https://electricityplans.com/texas/esid-lookup/

Reply:
That's interesting. Hopefully it's not possible to have two meters with the same number?

Maybe we should use essid_meternum?